vendor/github.com/containers/image/v5/docker/docker_transport.go

```diff
@@ -43,30 +43,44 @@ func (t dockerTransport) ValidatePolicyConfigurationScope(scope string) error {
 
 // dockerReference is an ImageReference for Docker images.
 type dockerReference struct {
-	ref reference.Named // By construction we know that !reference.IsNameOnly(ref)
+	ref             reference.Named // By construction we know that !reference.IsNameOnly(ref)
+	isUnknownDigest bool
 }
 
 // ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an Docker ImageReference.
 func ParseReference(refString string) (types.ImageReference, error) {
 	if !strings.HasPrefix(refString, "//") {
 		return nil, fmt.Errorf("docker: image reference %s does not start with //", refString)
 	}
+	// Check if ref has UnknownDigestSuffix suffixed to it
+	unknownDigest := false
+	if strings.HasSuffix(refString, reference.UnknownDigestSuffix) {
+		unknownDigest = true
+		refString = strings.TrimSuffix(refString, reference.UnknownDigestSuffix)
```

Comment:
In this case, the `ref` as parsed should be `IsNameOnly`, otherwise the input is inconsistent and should be rejected.

Reply:
Added a check for this